crates/uv-python/src/discovery.rs

```diff
@@ -1490,6 +1473,106 @@ impl PythonRequest {
         Self::ExecutableName(value.to_string())
     }
 
+    /// Try to parse a tool executable as a Python version, e.g. `uvx [executable]`.
+    ///
+    /// The `PythonRequest::parse` constructor above is intended for the `--python` flag, where the
+    /// value is unambiguously a Python version. This alternate constructor is intended for `uvx`,
+    /// where the executable might be a Python version or a package name. There are several
+    /// differences in behavior:
+    /// - This only supports long names, including e.g. `pypy39` but **not** `pp39` or `39`.
```

Comment:
The newline is necessary in some markdown renderers, let's include it for consistency
```suggestion
    ///
    /// - This only supports long names, including e.g. `pypy39` but **not** `pp39` or `39`.
```